### PR TITLE
codeintel: Run sqlite migrations in background

### PR DESCRIPTION
**Pull request by efritz**
On the start of the bundle manager, kick off a pool of goroutines that will open then immediately shut a reader backed by SQLite files on disk. This helps run migrations out of band in case larger files take too long to perform in the query path.

This will be polished more in subsequent PRs (not running PRs when unnecessary, better catching errors about SQLite exclusivity locks, etc).

### CODE REVIEW

File: cmd/precise-code-intel-bundle-manager/internal/readers/migrate.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+// of being paid on-demand when the database is opened within the query path. This method does not block
+// the startup of the bundle manager as it does not change the correctness of the service.
+func Migrate(bundleDir string) error {
+	paths, err := sqlitePaths(bundleDir)
+	if err != nil {
+		return err
+	}
+
+	ch := make(chan string, len(paths))
+	defer close(ch)
```
Comment (gbrik) [inline cmd/precise-code-intel-bundle-manager/internal/readers/migrate.go:31]: question:
errr, will this kill the process as soon as we return?

File: cmd/precise-code-intel-bundle-manager/internal/readers/migrate.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		}
+
+		sizeByPath[filename] = stat.Size()
+	}
+
+	// Construct a slice of the map keys
+	paths := make([]string, 0, len(sizeByPath))
+	for path := range sizeByPath {
+		paths = append(paths, path)
+	}
```
Comment (gbrik) [inline cmd/precise-code-intel-bundle-manager/internal/readers/migrate.go:94]: question:
how does `append` work? aka what are the asymptotics of this loop?

File: cmd/precise-code-intel-bundle-manager/internal/readers/migrate.go (hunk 1 above)
Comment (efritz) [reply cmd/precise-code-intel-bundle-manager/internal/readers/migrate.go:31]: No, this just signals "no more values" to the goroutines so they can exit.

File: cmd/precise-code-intel-bundle-manager/internal/readers/migrate.go (hunk 2 above)
Comment (efritz) [reply cmd/precise-code-intel-bundle-manager/internal/readers/migrate.go:94]: append grows the backing array by some multiplicative factor (1.5-2x, have to check source) if it doesn't have the capacity to add the element, but we're adding `len(sizeByPath)` items and that's the capacity we gave so each append is guaranteed to be `O(1)`.

**Pull request merged**